File: Leet Code/Convert Sorted List to Binary Search Tree/answer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insert(ArrayList<Integer> list, int low, int high){
        if(low <= high){
            int mid = low + (high-low) / 2;
            TreeNode newNode = new TreeNode(list.get(mid));
            newNode.left = insert(list, low, mid-1);
            newNode.right = insert(list, mid+1, high);

            return newNode;
        }
        return null;
    }
  
    public TreeNode sortedListToBST(ListNode head) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        while(head != null){
            list.add(head.val);
            head = head.next;
        }
        return insert(list, 0, list.size()-1);
    }
}

File: Leet Code/Range Sum of BST/answer.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum = 0;
    void traverse(TreeNode* root, int low, int high){
        if(root == NULL){
            return;
        }
        
        if(root->val >= low && root->val <= high){
            sum += root->val;
        }

        traverse(root->left, low, high);
        traverse(root->right, low, high);
    }
    int rangeSumBST(TreeNode* root, int low, int high) {
        traverse(root,low,high);
        return sum;
    }
};
